volatility/framework/plugins/mac/pslist.py

```diff
@@ -85,13 +121,120 @@ def list_tasks(cls,
             else:
                 seen[proc.vol.offset] = 1
 
-            if not filter_func(proc) and kernel_as.is_valid(proc.vol.offset, proc.vol.size):
+            if not filter_func(proc) and kernel_layer.is_valid(proc.vol.offset, proc.vol.size):
                 yield proc
 
             try:
                 proc = proc.p_list.le_next.dereference()
             except exceptions.InvalidAddressException:
```

Comment:
should this be PagedInvalidAddressException or InvalidAddressException? I notice this method uses InvalidAddressException whereas all the other enumeration methods use PagedInvalidAddressException. I assume this is how the plugins were before and you just moved the code over, but it seems like now is a good time to make consistent.

Reply:
So, if we only catch PagedInvalidAddressException, we don't get faults that might happen in lower layers.  If we do InvalidAddressException we catch any time an address is invalid?  Depends what you're after, but I agree consistency would be good, so once you've chosen I can make the changes throughout.

Reply:
Yep, unless there's an explicit reason to know which page or that it happened in the virtual layer, rather than a lower layer, we should be catching InvalidAddressException.  I've fixed that up (across the codebase) in here.